lib/auth/types/AuthType.js

```diff
@@ -112,7 +118,7 @@ export default class AuthType {
             options: {
                 authType: this.type,
                 authHeaderName: this.authHeaderName,
-                allowedAdditionalAuthHeaders: this.allowedAdditionalAuthHeaders,
+                allowedAuthHeaders: union(this.requestHeadersWhitelist, this.allowedAdditionalAuthHeaders),
```

Comment:
1. Since this variable does not store only Auth headers anymore, can you rename it to something else. Example: allowedHeaders, whitelistedHeaders, allowedRequestHeaders ...
2. Can you check how this code behaves when "requestHeadersWhitelist" is not set in kibana config ? Will this value be null/undefined and can it throw any exception in union(..) ?

Reply:
1. I will change it to allowedHeaders 
2. I checked the function of union in the lodash website https://lodash.com/docs/4.17.15#union (REPL) for null, undefined and empty values. They work as fine.

Reply:
2. I have added unit test for empty requestHeadersWhitelist object too